src/Microsoft.Health.Dicom.SqlServer/Features/Query/BaseSqlQueryGenerator.cs

```diff
@@ -117,6 +117,26 @@ public override void Visit(StringSingleValueMatchCondition stringSingleValueMatc
             .AppendLine();
     }
 
+    public override void Visit(StudyToSeriesStringSingleValueMatchCondition seriesStringSingleValueMatchCondition)
+    {
+        var queryTag = seriesStringSingleValueMatchCondition.QueryTag;
+        var studyDicomTagSqlEntry = DicomTagSqlEntry.GetStudyKeyDicomTagSqlEntry();
+        var seriesDicomTagSqlEntry = DicomTagSqlEntry.GetStudyToSeriesDicomTagSqlEntry(queryTag);
+
+        var tableAlias = GetTableAlias(studyDicomTagSqlEntry, null);
+        StringBuilder
+            .Append("AND ");
+        var seriesTableAlias = "ses";
+        StringBuilder
+            .Append(studyDicomTagSqlEntry.SqlColumn, tableAlias)
+            .Append(" IN  (SELECT DISTINCT StudyKey FROM Series " + seriesTableAlias + " WHERE ")
+            .Append(seriesDicomTagSqlEntry.SqlColumn, seriesTableAlias)
+            .Append("=")
+            .Append(_parameters.AddParameter(seriesDicomTagSqlEntry.SqlColumn, seriesStringSingleValueMatchCondition.Value))
+            .Append(")")
+            .AppendLine();
```

Comment:
Shouldn't this include partitionkey to reduce the results and use the index we have?
 `AND st.StudyKey IN  (SELECT DISTINCT StudyKey FROM Series ses WHERE ses.Modality=@p0 AND ses.PartitionKey = 1)`

Reply:
good point. Fixed